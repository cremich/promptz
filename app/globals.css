pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
