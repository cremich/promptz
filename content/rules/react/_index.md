---
title: "React"
description: "Rules for React development, component patterns, and best practices."
---
