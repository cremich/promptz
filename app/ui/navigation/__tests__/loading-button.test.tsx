import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import LoadingButton from "@/app/ui/navigation/loading-button";

describe("Loading Button", () => {
  test("renders loading button unchanged", () => {
    const { container } = render(<LoadingButton />);
    expect(container).toMatchSnapshot();
  });
});
