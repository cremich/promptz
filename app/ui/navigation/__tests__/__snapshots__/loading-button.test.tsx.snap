// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Button renders loading button unchanged 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
    disabled=""
  >
    <svg
      class="lucide lucide-loader-circle animate-spin"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
    Please wait
  </button>
</div>
`;
